python-packages/json_schemas/src/zero_ex/json_schemas/__init__.py

```diff
@@ -6,26 +6,33 @@
 
 from pkg_resources import resource_string
 import jsonschema
+from stringcase import snakecase
 
 
 class _LocalRefResolver(jsonschema.RefResolver):
     """Resolve package-local JSON schema id's."""
 
     def __init__(self):
         """Initialize a new instance."""
-        self.ref_to_file = {
-            "/addressSchema": "address_schema.json",
-            "/hexSchema": "hex_schema.json",
-            "/orderSchema": "order_schema.json",
-            "/wholeNumberSchema": "whole_number_schema.json",
-            "/ECSignature": "ec_signature_schema.json",
-            "/signedOrderSchema": "signed_order_schema.json",
-            "/ecSignatureParameterSchema": (
-                "ec_signature_parameter_schema.json" + ""
-            ),
-        }
         jsonschema.RefResolver.__init__(self, "", "")
 
+    @staticmethod
+    def _ref_to_file(ref: str) -> str:
+        """Translate a JSON schema ref to its corresponding file name.
+
+        >>> _LocalRefResolver._ref_to_file("/addressSchema")
+        'address_schema.json'
+        """
+        _ref = ref.lstrip("/")
+
+        # handle weird special cases
+        _ref = _ref.replace("ECSignature", "EcSignature")
```

Comment:
Is this an inconsistency in our schema ref names? We should fix this in the JS package, rather then special case this here.

Reply:
I was thinking that this wasn't an inconsistency, but rather a truly special case, analogous to how the ERC20.sol's generated TypeScript wrappers would get turned into e_r_c_20.ts without intervention.

But looking further now I see that we have a convention of not fully uppercasing acronyms in schema ref ids (eg `/eip712TypedDataSchema` and `/ecSignatureParameterSchema`).  So yes, I'd say it makes sense to change ref id `/ECSignature` to `/ecSignature`.  Does that make sense to you?

But also, now that you mention it, there is also a real inconsistency here that we could address.  It's the second special case in this code block.  It's that most schema refs (37/44) end with the word "Schema", but some (8/44*) don't:

```
schemas$ grep -He 'id":' * | grep -v Schema
asset_pairs_request_opts_schema.json:    "id": "/AssetPairsRequestOpts",
ec_signature_schema.json:    "id": "/ECSignature",
js_number.json:    "id": "/jsNumber",
orderbook_request_schema.json:    "id": "/OrderBookRequest",
order_config_request_schema.json:    "id": "/OrderConfigRequest",
orders_request_opts_schema.json:    "id": "/OrdersRequestOpts",
paged_request_opts_schema.json:    "id": "/PagedRequestOpts",
request_opts_schema.json:    "id": "/RequestOpts",
schemas$ grep -He 'id":' * | grep -v Schema | wc -l
8
```

Also, I see that while most files (43/44) end with `_schema.json`, there's one (`js_number.json`) that doesn't!  That would have broken my code here!

If we fix all of these things in `@0x/json-schemas` then this method will collapse to just `return f"{snakecase(ref.lstrip("/"))}_schema.json"`!  I'll do it.

Reply:
Yep, awesome, let's make both of those changes to @0x/json-schemas.